builder/alicloud/ecs/step_attach_keypair.go

```diff
@@ -21,7 +23,7 @@ func (s *stepAttachKeyPair) Run(_ context.Context, state multistep.StateBag) mul
 	config := state.Get("config").(Config)
```

Comment:
This config should be a pointer so that you don't have to add a field to `stepAttachKeyPair` . 

`config := state.Get("config").(*Config)`